### ISSUE TITLE
Ping embedded devices

### ISSUE THREAD

**Issue by multiflexi** [Ylianst/MeshCentral#1469]
Hi,
would it be possible to add devices which cannot run the client just by IP and MeshCentral would check if the IP is alive by ping? This would be great for embedded devices like Epson projectors. They have only embedded Linux without user access. The on-line check is possible either by using regular ping or [PJlink](https://pjlink.jbmia.or.jp/english/data_cl2/PJLink_5-1.pdf) which enables control of power and other things.

**Comment by Ylianst** [comment]
Well, I have no idea what PJlink was, that is interesting, I wonder how many devices made use of that specification.  You ask would only be useful for people setting up MeshCentral in "LAN mode" since MeshCentral would need to ping devices on the local network? I am not sure how many people use MeshCentral in that mode, it's certainly where almost everyone starts.

You asking to be able to add a dummy device on MeshCentral that would do nothing by ping for the presence of the device? If I understand correctly, you would not be able to do anything else?

**Comment by multiflexi** [comment]
Yes, I thought one could add a device by its IP and MeshCentral would just check if it is alive, maybe show additional info which can be discovered over network, but that is just nice to have. We use it in LAN only mode for administration of RPis and PCs in an interactive gallery and there are some embedded devices like projectors and it would be nice to see them in MeshCentral too. Simple way is regular ping, more advanced way would be PJlink which also allows scheduled power control, AV control, some monitoring (e. g. lamp life) etc. PJlink is used by many projectors and displays from companies like Panasonic, Epson, Nec, BenQ, Sony, Canon or Optoma. Here you can find [introduction](https://pjlink.jbmia.or.jp/english/data/6-1_PJLink%20Introduction_20180622.pdf) and [documentation](https://pjlink.jbmia.or.jp/english/dl_class2.html). I noticed there is npm package [pjlink](https://github.com/sy1vain/node-pjlink)

**Comment by muahdib69** [comment]
Hello Multiflexi, 

I just read your comments and you might be better of using librenms as a monitoring tool, which can detect nearly anything on your network and graph and plot a lot of device information and is also free. 

And I am pretty sure you could embed in librenms in the devicepage access to meshcentral functionality ... 

At least that would be the way I would be going in your situation, as adding ping monitoring functionality would leave you with a half-backed monitoring solution, because mesh-central is not envisioned as monitoring devices (no RRD graphs, no SNMP gathering, etc.)

But if Ylianst is willing to implement it, great :-)

**Comment by multiflexi** [comment]
Hi, graphs and SNMP are not necessary, just simple check if the device is alive, the same way you see devices with MC agents and because you cannot install agent on the device I suggested the control over PJlink, MC is quite user friendly and I think it is better to use just one app for the same thing, than two or three, especially when the user is that tech savvy.
